From ed8ccce0d343961640c232e35c5bbd9cb4ea510f Mon Sep 17 00:00:00 2001
From: Beni Lev <beni.lev@intel.com>
Date: Tue, 18 Jun 2013 15:26:32 +0300
Subject: Avoid FW reload on SoftAP enablement

BRCM are using an other uCode for SoftAP, but we do not,
so just skip the FW switch

Category: Device Enablement
Upstream Candidate: Yes
Domain: AOSP-wifi
Issue: ABRO-114, ABRO-151
Change-Id: If531db2e1d65c855a855593dc37ce3520f759922
Signed-off-by: Beni Lev <beni.lev@intel.com>
---
 Android.mk           |    5 +++++
 SoftapController.cpp |    5 +++++
 2 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/Android.mk b/Android.mk
index da4a7ad..3a04e75 100644
--- a/Android.mk
+++ b/Android.mk
@@ -38,6 +38,11 @@ LOCAL_C_INCLUDES := $(KERNEL_HEADERS) \
 
 LOCAL_CFLAGS := -Werror=format
 
+ifeq ($(NO_FW_RELOAD_FOR_SOFTAP),true)
+ LOCAL_CFLAGS +=  \
+               -DNO_FW_RELOAD_FOR_SOFTAP
+endif
+
 LOCAL_SHARED_LIBRARIES := libstlport libsysutils liblog libcutils libnetutils \
                           libcrypto libhardware_legacy libmdnssd libdl \
                           liblogwrap
diff --git a/SoftapController.cpp b/SoftapController.cpp
index 825a376..a482784 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -201,6 +201,10 @@ int SoftapController::setSoftap(int argc, char *argv[]) {
  */
 int SoftapController::fwReloadSoftap(int argc, char *argv[])
 {
+#ifdef NO_FW_RELOAD_FOR_SOFTAP
+    ALOGD("Softap skip FW reload");
+    return 0;
+#else
     int i = 0;
     char *fwpath = NULL;
 
@@ -226,6 +230,7 @@ int SoftapController::fwReloadSoftap(int argc, char *argv[])
         ALOGD("Softap fwReload - Ok");
     }
     return ResponseCode::SoftapStatusResult;
+#endif
 }
 
 void SoftapController::generatePsk(char *ssid, char *passphrase, char *psk_str) {
-- 
1.7.4.1

